### PR TITLE
raftstore: use force in compact_range triggered by no valid split key

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Close #15282

What's Changed:

```commit-message
use option bottommost_level_compaction = force in the compact_range scheduled by no_valid_split_key.
```
In no_valid_split_key senario, we schedule a compact range for the given region. We shoud also set bottommost_level_compaction = force to avoid the trivial move in the bottommost level so that tombstones can be removed..

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by tonyxuqqi — APPROVED** [review]
I think it's better to have a CompactionOption as parameter instead of 3 parameters.

File: components/raftstore/src/store/worker/compact.rs [hunk 1]
```diff
@@ -301,16 +304,19 @@ where
...
-        );
+        box_try!(self.engine.compact_range_cf(
+            cf_name,
+            start_key,
+            end_key,
+            ManualCompactionOptions::new(false, 1, bottommost_level_force),
+        ));
         compact_range_timer.observe_duration();
         info!(
             "compact range finished";
```
Comment (nolouch) [inline components/raftstore/src/store/worker/compact.rs:322]: maybe we can add the filed in the log about `compact_option`.

**Review by nolouch — APPROVED** [review]
lgtm

File: components/raftstore/src/store/worker/compact.rs (hunk 1 above)
Comment (SpadeA-Tang) [reply components/raftstore/src/store/worker/compact.rs:322]: fine

**Pull request merged**